modules/itc/src/main/scala/lucuma/itc/ExposureTime.scala

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2016-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.itc
+
+import cats.syntax.all.*
+import eu.timepit.refined.types.numeric.PosInt
+import io.circe.*
+import io.circe.syntax.*
+import lucuma.core.math.SignalToNoise
+import lucuma.core.model.NonNegDuration
+import lucuma.core.util.Enumerated
+import lucuma.itc.encoders.given
+import lucuma.itc.search.ObservingMode
+
+import scala.concurrent.duration.FiniteDuration
+
+sealed trait ExposureTimeResult extends Product with Serializable
```

Comment:
Maybe a `sealed trait ExposureTimeError extends ExposureTimeResult` to group `CalculationError` and `SourceTooBright`. Thinking about it a bit more, I think it might help future proof the schema to have this in the schema as well.  Something like

```
interface ExposureTimeError {
  message: String!
}

union ExposureTimeResult = ExposureEstimate | SourceTooBright | CalculationError

type ExposureEstimate {
   # Exposure time duration
   exposureTime: Duration!

   # Exposures
   exposures: PosInt!

   # Signal/Noise ratio
   signalToNoise: SignalToNoise!
 }

 # ITC error when the source is too bright
 type SourceTooBright implements ExposureTimeError {
   message: String!

   # Half-well timme
   halfWellTime: BigDecimal!
 }

 # ITC generic error result
 type CalculationError implements ExposureTimeError {
  message: String!

   # Messages, the itc engine sometimes returns multiple error lines
   details: [String!]!
 }
```

That way a query that only cares about success vs. failure could "match" `... on ExposureTimeError` and not the individual cases.  Or you can add that and not worry about new error cases coming up in the future.

Reply:
Sounds very good, Would the union then be 
`union ExposureTimeResult = ExposureEstimate | ExposureTimeError`?

Reply:
Hmm, I don't know.  I think the `union` should still list the individual result cases. If you don't, I'm not sure you can then still match `... on CalculationError` specifically if you wanted to.